tutorials/src/bin/basic-tutorial-4.rs

```diff
@@ -13,7 +13,17 @@ struct CustomData {
     duration: gst::ClockTime, // How long does this media last, in nanoseconds
 }
 
+#[path="../tutorials-common.rs"]
+mod tutorials_common;
+
 fn main() {
+    // tutorials_common::run is only required to set up the application environment on macOS
+    // (but not necessary in normal Cocoa applications where this is set up automatically)
+    tutorials_common::run(tutorial_main);
```

Comment:
I would do this slightly different (also don't copy the explanation into every tutorial, make it just a line :) ). I would make it some kind of `tutorials_common::init()` that is the first thing in main. Doing it with a closure that calls the actual main is nice, but it adds more compliciation to the tutorial IMHO

The same change is also needed for everything in the examples subdirectory, for everything that is showing video. Also tutorial 7 needs this too.

Reply:
Actually ignore this comment. You're moving the actual tutorial code in a different thread. Because the NSRunLoop has to run on the main thread.

Then we only need to add this also to tutorial 7 and the examples.